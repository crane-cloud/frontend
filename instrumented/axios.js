function cov_12zl524xrh() {
  var path = "/home/greatest/Desktop/code/cranecloud/frontend/em/frontend/src/axios.js";
  var hash = "4ba0e69af2be53c2e422da5c94744e69b996097f";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/home/greatest/Desktop/code/cranecloud/frontend/em/frontend/src/axios.js",
    statementMap: {
      "0": {
        start: {
          line: 4,
          column: 17
        },
        end: {
          line: 6,
          column: 2
        }
      },
      "1": {
        start: {
          line: 8,
          column: 0
        },
        end: {
          line: 10,
          column: 4
        }
      },
      "2": {
        start: {
          line: 12,
          column: 0
        },
        end: {
          line: 26,
          column: 2
        }
      },
      "3": {
        start: {
          line: 13,
          column: 16
        },
        end: {
          line: 13,
          column: 24
        }
      },
      "4": {
        start: {
          line: 15,
          column: 12
        },
        end: {
          line: 15,
          column: 17
        }
      },
      "5": {
        start: {
          line: 17,
          column: 4
        },
        end: {
          line: 23,
          column: 5
        }
      },
      "6": {
        start: {
          line: 18,
          column: 6
        },
        end: {
          line: 18,
          column: 27
        }
      },
      "7": {
        start: {
          line: 19,
          column: 6
        },
        end: {
          line: 19,
          column: 33
        }
      },
      "8": {
        start: {
          line: 20,
          column: 11
        },
        end: {
          line: 23,
          column: 5
        }
      },
      "9": {
        start: {
          line: 21,
          column: 6
        },
        end: {
          line: 21,
          column: 56
        }
      },
      "10": {
        start: {
          line: 22,
          column: 6
        },
        end: {
          line: 22,
          column: 41
        }
      },
      "11": {
        start: {
          line: 24,
          column: 4
        },
        end: {
          line: 24,
          column: 29
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 13,
            column: 2
          },
          end: {
            line: 13,
            column: 3
          }
        },
        loc: {
          start: {
            line: 13,
            column: 16
          },
          end: {
            line: 13,
            column: 24
          }
        },
        line: 13
      },
      "1": {
        name: "(anonymous_1)",
        decl: {
          start: {
            line: 14,
            column: 2
          },
          end: {
            line: 14,
            column: 3
          }
        },
        loc: {
          start: {
            line: 14,
            column: 13
          },
          end: {
            line: 25,
            column: 3
          }
        },
        line: 14
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        }, {
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        }],
        line: 17
      },
      "1": {
        loc: {
          start: {
            line: 17,
            column: 8
          },
          end: {
            line: 17,
            column: 70
          }
        },
        type: "binary-expr",
        locations: [{
          start: {
            line: 17,
            column: 8
          },
          end: {
            line: 17,
            column: 37
          }
        }, {
          start: {
            line: 17,
            column: 41
          },
          end: {
            line: 17,
            column: 70
          }
        }],
        line: 17
      },
      "2": {
        loc: {
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        }, {
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        }],
        line: 20
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0,
      "6": 0,
      "7": 0,
      "8": 0,
      "9": 0,
      "10": 0,
      "11": 0
    },
    f: {
      "0": 0,
      "1": 0
    },
    b: {
      "0": [0, 0],
      "1": [0, 0],
      "2": [0, 0]
    },
    _coverageSchema: "1a1c01bbd47fc00a2c39e90264f33305004495a9",
    hash: "4ba0e69af2be53c2e422da5c94744e69b996097f"
  };
  var coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  var actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_12zl524xrh = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_12zl524xrh();
import axios from "axios";
import { API_BASE_URL } from "./config";
const instance = (cov_12zl524xrh().s[0]++, axios.create({
  baseURL: API_BASE_URL
}));
cov_12zl524xrh().s[1]++;
instance.defaults.headers.Authorization = `Bearer ${localStorage.getItem("token")}`;
cov_12zl524xrh().s[2]++;
instance.interceptors.response.use(response => {
  cov_12zl524xrh().f[0]++;
  cov_12zl524xrh().s[3]++;
  return response;
}, error => {
  cov_12zl524xrh().f[1]++;
  let e = (cov_12zl524xrh().s[4]++, error);
  // this checks for when a token is not verified and logs you out
  cov_12zl524xrh().s[5]++;
  if ((cov_12zl524xrh().b[1][0]++, error.response.status === 401) || (cov_12zl524xrh().b[1][1]++, error.response.status === 422)) {
    cov_12zl524xrh().b[0][0]++;
    cov_12zl524xrh().s[6]++;
    localStorage.clear();
    cov_12zl524xrh().s[7]++;
    window.location.href = "/";
  } else {
    cov_12zl524xrh().b[0][1]++;
    cov_12zl524xrh().s[8]++;
    if (error.response.status === 502) {
      cov_12zl524xrh().b[2][0]++;
      cov_12zl524xrh().s[9]++;
      e = "Your request too long possible Server Error.";
      cov_12zl524xrh().s[10]++;
      window.location.href = "/projects";
    } else {
      cov_12zl524xrh().b[2][1]++;
    }
  }
  cov_12zl524xrh().s[11]++;
  return Promise.reject(e);
});
export default instance;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfMTJ6bDUyNHhyaCIsImFjdHVhbENvdmVyYWdlIiwiYXhpb3MiLCJBUElfQkFTRV9VUkwiLCJpbnN0YW5jZSIsInMiLCJjcmVhdGUiLCJiYXNlVVJMIiwiZGVmYXVsdHMiLCJoZWFkZXJzIiwiQXV0aG9yaXphdGlvbiIsImxvY2FsU3RvcmFnZSIsImdldEl0ZW0iLCJpbnRlcmNlcHRvcnMiLCJyZXNwb25zZSIsInVzZSIsImVycm9yIiwiZiIsImUiLCJiIiwic3RhdHVzIiwiY2xlYXIiLCJ3aW5kb3ciLCJsb2NhdGlvbiIsImhyZWYiLCJQcm9taXNlIiwicmVqZWN0Il0sInNvdXJjZXMiOlsiYXhpb3MuanMiXSwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGF4aW9zIGZyb20gXCJheGlvc1wiO1xuaW1wb3J0IHsgQVBJX0JBU0VfVVJMIH0gZnJvbSBcIi4vY29uZmlnXCI7XG5cbmNvbnN0IGluc3RhbmNlID0gYXhpb3MuY3JlYXRlKHtcbiAgYmFzZVVSTDogQVBJX0JBU0VfVVJMLFxufSk7XG5cbmluc3RhbmNlLmRlZmF1bHRzLmhlYWRlcnMuQXV0aG9yaXphdGlvbiA9IGBCZWFyZXIgJHtsb2NhbFN0b3JhZ2UuZ2V0SXRlbShcbiAgXCJ0b2tlblwiXG4pfWA7XG5cbmluc3RhbmNlLmludGVyY2VwdG9ycy5yZXNwb25zZS51c2UoXG4gIChyZXNwb25zZSkgPT4gcmVzcG9uc2UsXG4gIChlcnJvcikgPT4ge1xuICAgIGxldCBlID0gZXJyb3I7XG4gICAgLy8gdGhpcyBjaGVja3MgZm9yIHdoZW4gYSB0b2tlbiBpcyBub3QgdmVyaWZpZWQgYW5kIGxvZ3MgeW91IG91dFxuICAgIGlmIChlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDQwMSB8fCBlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDQyMikge1xuICAgICAgbG9jYWxTdG9yYWdlLmNsZWFyKCk7XG4gICAgICB3aW5kb3cubG9jYXRpb24uaHJlZiA9IFwiL1wiO1xuICAgIH0gZWxzZSBpZihlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDUwMil7XG4gICAgICBlID0gXCJZb3VyIHJlcXVlc3QgdG9vIGxvbmcgcG9zc2libGUgU2VydmVyIEVycm9yLlwiXG4gICAgICB3aW5kb3cubG9jYXRpb24uaHJlZiA9IFwiL3Byb2plY3RzXCI7XG4gICAgfVxuICAgIHJldHVybiBQcm9taXNlLnJlamVjdChlKTtcbiAgfVxuKTtcblxuZXhwb3J0IGRlZmF1bHQgaW5zdGFuY2U7XG4iXSwibWFwcGluZ3MiOiI7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7OztJQWVZO0lBQUFBLGNBQUEsWUFBQUEsQ0FBQTtNQUFBLE9BQUFDLGNBQUE7SUFBQTtFQUFBO0VBQUEsT0FBQUEsY0FBQTtBQUFBO0FBQUFELGNBQUE7QUFmWixPQUFPRSxLQUFLLE1BQU0sT0FBTztBQUN6QixTQUFTQyxZQUFZLFFBQVEsVUFBVTtBQUV2QyxNQUFNQyxRQUFRLElBQUFKLGNBQUEsR0FBQUssQ0FBQSxPQUFHSCxLQUFLLENBQUNJLE1BQU0sQ0FBQztFQUM1QkMsT0FBTyxFQUFFSjtBQUNYLENBQUMsQ0FBQztBQUFDSCxjQUFBLEdBQUFLLENBQUE7QUFFSEQsUUFBUSxDQUFDSSxRQUFRLENBQUNDLE9BQU8sQ0FBQ0MsYUFBYSxHQUFJLFVBQVNDLFlBQVksQ0FBQ0MsT0FBTyxDQUN0RSxPQUFPLENBQ1AsRUFBQztBQUFDWixjQUFBLEdBQUFLLENBQUE7QUFFSkQsUUFBUSxDQUFDUyxZQUFZLENBQUNDLFFBQVEsQ0FBQ0MsR0FBRyxDQUMvQkQsUUFBUSxJQUFLQTtFQUFBQSx1QkFBQTtFQUFBQSx1QkFBQTtFQUFBQSxNQUFBLENBQUFBLFFBQVE7QUFBRCxDQUFDLEVBQ3JCRSxLQUFLLElBQUs7RUFBQWhCLGNBQUEsR0FBQWlCLENBQUE7RUFDVCxJQUFJQyxDQUFDLElBQUFsQixjQUFBLEdBQUFLLENBQUEsT0FBR1csS0FBSztFQUNiO0VBQUFoQixjQUFBLEdBQUFLLENBQUE7RUFDQSxJQUFJLENBQUFMLGNBQUEsR0FBQW1CLENBQUEsVUFBQUgsS0FBSyxDQUFDRixRQUFRLENBQUNNLE1BQU0sS0FBSyxHQUFHLE1BQUFwQixjQUFBLEdBQUFtQixDQUFBLFVBQUlILEtBQUssQ0FBQ0YsUUFBUSxDQUFDTSxNQUFNLEtBQUssR0FBRyxHQUFFO0lBQUFwQixjQUFBLEdBQUFtQixDQUFBO0lBQUFuQixjQUFBLEdBQUFLLENBQUE7SUFDbEVNLFlBQVksQ0FBQ1UsS0FBSyxFQUFFO0lBQUNyQixjQUFBLEdBQUFLLENBQUE7SUFDckJpQixNQUFNLENBQUNDLFFBQVEsQ0FBQ0MsSUFBSSxHQUFHLEdBQUc7RUFDNUIsQ0FBQyxNQUFNO0lBQUF4QixjQUFBLEdBQUFtQixDQUFBO0lBQUFuQixjQUFBLEdBQUFLLENBQUE7SUFBQSxJQUFHVyxLQUFLLENBQUNGLFFBQVEsQ0FBQ00sTUFBTSxLQUFLLEdBQUcsRUFBQztNQUFBcEIsY0FBQSxHQUFBbUIsQ0FBQTtNQUFBbkIsY0FBQSxHQUFBSyxDQUFBO01BQ3RDYSxDQUFDLEdBQUcsOENBQThDO01BQUFsQixjQUFBLEdBQUFLLENBQUE7TUFDbERpQixNQUFNLENBQUNDLFFBQVEsQ0FBQ0MsSUFBSSxHQUFHLFdBQVc7SUFDcEMsQ0FBQztNQUFBeEIsY0FBQSxHQUFBbUIsQ0FBQTtJQUFBO0VBQUQ7RUFBQ25CLGNBQUEsR0FBQUssQ0FBQTtFQUNELE9BQU9vQixPQUFPLENBQUNDLE1BQU0sQ0FBQ1IsQ0FBQyxDQUFDO0FBQzFCLENBQUMsQ0FDRjtBQUVELGVBQWVkLFFBQVEifQ==