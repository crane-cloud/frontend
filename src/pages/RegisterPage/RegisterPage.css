.RegisterPageContainer {
  background: var(--primary-color) !important;
  color: #ffffff;
  display: flex;
  flex-direction: column;
  flex: 1;
}

.RegisterContent {
  display: flex;
  flex-direction: column;
  align-self: center;
  justify-content: center;
  flex: 1;
  padding-bottom: 5rem;
}

.RegisterContentHeading {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.RegisterSuccessContent {
  min-height: 25rem;
  display: grid;
  grid-template-columns: 1fr;
  justify-content: center;
  align-items: center;
}

.RegisterSuccessContent h2 {
  color: var(--secondary-color);
}

.RegisterSuccessContent p span {
  color: var(--secondary-color);
}

.RegisterTerms {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  gap: 1rem;
  padding: 1rem 0;
}

.RegisterErrorDiv {
  background-color: rgba(255, 255, 255, 0.5);
  color: #d63b3b;
  text-align: center;
  padding: 1rem 1rem;
  border-radius: 5px;
}

.LogoIcon {
  width: 1.5rem;
  height: 1.5rem;
}

.GitText {
  color: #fff;
  text-transform: capitalize;
}

.LowerSignupSection {
  display: grid;
  grid-template-columns: 1fr;
  padding-top: 0.5rem;
  gap: 0.5rem;
}

@media only screen and (max-width: 640px) {
  /* .RegisterContent {
    width: 90%;
  }

  .InputPassword,
  .InputText {
    width: 90%;
  }

  .VerificationCodeInput {
    width: 90%;
  }

  .VerificationCodeInput .InputText {
    width: 100%;
  } */
}

.password-wrapper {
  position: relative;
  display: grid;
  align-items: center;
}

.password-wrapper .toggle-password {
  position: absolute;
  top: 50%;
  right: 1rem;
  transform: translateY(-50%);
  cursor: pointer;
}

.password-input-container {
  position: relative;
}

.password {
  position: absolute;
  top: 50%;
  right: 1rem;
  transform: translateY(-50%);
  cursor: pointer;
}

.password-repeat {
  position: relative;
  display: grid;
  align-items: center;
}

.passwordConfirm.toggle-password {
  position: absolute;
  top: 50%;
  right: 1rem;
  transform: translateY(-50%);
  cursor: pointer;
}

.passwordConfirm {
  position: absolute;
  top: 50%;
  right: 1rem;
  transform: translateY(-50%);
  cursor: pointer;
}