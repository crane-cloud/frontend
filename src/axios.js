import axios from "axios";
import { API_BASE_URL, ACTIVITY_LOGS_API_URL, MONITORING_API_URL } from "./config";



// Create main backend axios instance
const instance = axios.create({
  baseURL: API_BASE_URL,
});
// Create activity axios instance
const userActivityLoggerAxios = axios.create({
  baseURL: ACTIVITY_LOGS_API_URL,
});

// Create monitoring axios instance
const monitoringAxios = axios.create({
  baseURL: MONITORING_API_URL,
});

// Set default headers
const token = localStorage.getItem("token");
instance.defaults.headers.Authorization = `Bearer ${token}`;
userActivityLoggerAxios.defaults.headers.Authorization = `Bearer ${token}`;
monitoringAxios.defaults.headers.Authorization = `Bearer ${token}`;

// Define response interceptor function
const responseInterceptor = (response) => response;

const errorInterceptor = (error) => {
  let e = error;
  // this checks for when a token is not verified and logs you out
  if (error.response.status === 401 || error.response.status === 422) {
    localStorage.clear();
    window.location.href = "/";
  } else if (error.response.status === 502) {
    e = "Your request took too long, possibly a Server Error.";
    window.location.href = "/projects";
  }
  return Promise.reject(e);
};

// Add response interceptors to both instances
instance.interceptors.response.use(responseInterceptor, errorInterceptor);
userActivityLoggerAxios.interceptors.response.use(
  responseInterceptor,
  errorInterceptor
);
monitoringAxios.interceptors.response.use(
  responseInterceptor,
  errorInterceptor
);

export { instance, userActivityLoggerAxios, monitoringAxios };
export default instance;
