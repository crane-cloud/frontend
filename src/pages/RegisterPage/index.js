import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import Header from "../../components/Header";
import InputText from "../../components/InputText";
import PrimaryButton from "../../components/PrimaryButton";
import Spinner from "../../components/Spinner";
import { ReactComponent as LogoIcon } from "../../assets/images/githublogo.svg";
import { API_BASE_URL, GIT_REDIRECT_URL } from "../../config";
import Checkbox from "../../components/Checkbox";
import { ReactComponent as Open } from "../../assets/images/open.svg";
import { ReactComponent as Closed } from "../../assets/images/close.svg";

import "./RegisterPage.css";

export default class RegisterPage extends Component {
  constructor() {
    super();

    this.state = {
      name: "",
      username: "",
      organisation:"",
      email: "",
      password: "",
      passwordConfirm: "",
      displayPassword: false,
      hasAgreed: false,
      loading: false,
      registered: false,
      gitLoading: false,
      error: "",
      passwordShown: false,
      passwordChecked: false,
    };

    this.handleOnChange = this.handleOnChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.toggleAgreed = this.toggleAgreed.bind(this);
    this.validateEmail = this.validateEmail.bind(this);
    this.toGithubauth = this.toGithubauth.bind(this);
    this.togglePassword = this.togglePassword.bind(this);
    this.togglePasswordConfirm = this.togglePasswordConfirm.bind(this);
  }

  toggleAgreed() {
    const { hasAgreed, error } = this.state;
    this.setState({
      hasAgreed: !hasAgreed,
    });

    if (error) {
      this.setState({
        error: "",
      });
    }
  }
  toGithubauth = () => {
    // on return, github will be redricted to the login page where the
    // functions that handle the rest of the authrntications are
    const { hasAgreed } = this.state;
    if (hasAgreed) {
      this.setState({
        loading: true,
        gitLoading: true,
      });
      window.location.href = `${GIT_REDIRECT_URL}`;
    } else {
      this.setState({
        loading: false,
        gitLoading: false,
        error: "Please agree to our Terms of Service",
      });
    }
  };

  handleOnChange(e) {
    const { error } = this.state;
    this.setState({
      [e.target.name]: e.target.value,
    });

    if (error) {
      this.setState({
        error: "",
      });
    }
  }

  validateEmail(email) {
    const emailRegEx =
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return emailRegEx.test(String(email).toLowerCase());
  }
  togglePassword() {
    this.setState({ passwordShown: !this.state.passwordShown });
  }
  togglePasswordConfirm() {
    this.setState({ displayPassword: !this.state.displayPassword });
  }

  handleSubmit(e) {
    e.preventDefault();

    const { name, username, organisation, email, password, passwordConfirm, hasAgreed } =
      this.state;

    const userData = {
      name,
      username,
      email,
      password,
    };

    if (!email || !password || !name || !username || !passwordConfirm) {
      this.setState({
        error: "Please enter all fields",
      });
    } else if (!name.trim()) {
      this.setState({
        error: "Please provide a valid username",
      });
    } else if (this.validateEmail(email) === false) {
      this.setState({
        loading: false,
        error: "Please provide a valid email address",
      });
    } else if (password !== passwordConfirm) {
      this.setState({
        loading: false,
        error: "Passwords do not match",
      });
    } else if (!hasAgreed) {
      this.setState({
        loading: false,
        error: "Please agree to our Terms of Service",
      });
    } else {
      this.setState({
        loading: true,
      });
      axios
        .post(`${API_BASE_URL}/users`, userData)
        .then((response) => {
          if (response.data.status === "success") {
            this.setState({
              loading: false,
              registered: true,
            });
          }
        })
        .catch((error) => {
          this.setState({
            loading: false,
            error: "Email already in use by another account.",
          });
        });
    }
  }

  render() {
    const {
      name,
      email,
      password,
      passwordConfirm,
      loading,
      registered,
      passwordShown,
      displayPassword,
      username,
      organisation,
      error,
      hasAgreed,
      gitLoading,
    } = this.state;
    return (
      <div className="RegisterPageContainer">
        <Header />
        <div className="RegisterContent">
          {!registered ? (
            <div>
              <div className="RegisterContentHeading">
                <h1>Create an account</h1>
              </div>
              <form onSubmit={this.handleSubmit} className="LoginContentInputs">
                <InputText
                  required
                  placeholder="Name"
                  name="name"
                  value={name}
                  onChange={this.handleOnChange}
                />
                <InputText
                  required
                  placeholder="Username"
                  name="username"
                  value={username}
                  onChange={this.handleOnChange}
                />
                <InputText
                  required
                  placeholder="Organisation"
                  name="organisation"
                  value={organisation}
                  onChange={this.handleOnChange}
                />
                <InputText
                  required
                  placeholder="Email Address"
                  name="email"
                  type="email"
                  value={email}
                  onChange={this.handleOnChange}
                />
                <div className="password-wrapper">
                  <InputText
                    required
                    placeholder="Password"
                    name="password"
                    type={passwordShown ? "text" : "password"}
                    value={password}
                    onChange={this.handleOnChange}
                  />

                  <div className="password" onClick={this.togglePassword}>
                    {passwordShown ? <Open /> : <Closed />}
                  </div>
                </div>
                <div className="password-repeat">
                  <InputText
                    required
                    placeholder="Repeat Password"
                    name="passwordConfirm"
                    type={displayPassword ? "text" : "password"}
                    value={passwordConfirm}
                    onChange={this.handleOnChange}
                  />
                  <div
                    className="password"
                    onClick={this.togglePasswordConfirm}
                  >
                    {displayPassword ? <Open /> : <Closed />}
                  </div>
                </div>

                {error && <div className="RegisterErrorDiv">{error}</div>}
                <div className=" RegisterTerms">
                  <Checkbox onClick={this.toggleAgreed} isChecked={hasAgreed} />
                  <div>
                    I agree to Crane Cloud{" "}
                    <Link
                      to="/terms-of-service"
                      target="_blank"
                      className="LoginContentLink"
                    >
                      Terms of service.
                    </Link>
                  </div>
                </div>

                <PrimaryButton
                  className="LoginButton AuthBtn"
                  type="submit"
                  onClick={this.handleSubmit}
                >
                  {loading ? <Spinner /> : "Register"}
                </PrimaryButton>
              </form>
              <div className="LowerLoginSection">
                <div>
                  <p className="LoginWith">
                    <span>Or join with</span>
                  </p>
                </div>
                <PrimaryButton
                  className="GithubLoginBtn AuthBtn"
                  disabled={gitLoading}
                  onClick={this.toGithubauth}
                >
                  {gitLoading ? (
                    <Spinner />
                  ) : (
                    <div className="GitLoginBtn">
                      <LogoIcon className="LogoIcon" />
                      <div className="GitText">Github</div>
                    </div>
                  )}
                </PrimaryButton>
              </div>
              <div className="LoginContentBottomLink LoginLinkContainer">
                Already have an account? &nbsp;
                <Link to="/login" className="LoginContentLink">
                  Go to Login
                </Link>
              </div>
            </div>
          ) : (
            <div className="RegisterSuccessContent">
              <div className="RegisteredMessage">
                <h2>Thank you for registering with us!</h2>
                <p>
                  We&apos;ve sent a link to your email address:&nbsp;
                  <span>{email}</span>
                  .
                  <br />
                  <br />
                  The link will expire after 24 hours. Please use this link to
                  activate and start using your account.
                </p>
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
}
