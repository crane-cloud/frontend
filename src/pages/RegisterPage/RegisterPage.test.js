import React from "react";
import { shallow } from "enzyme";
import RegisterPage from "./index";
import axios from "axios";

describe("<RegisterPage/> Component", () => {
  let RegisterPageComponent;

  beforeEach(() => {
    RegisterPageComponent = shallow(<RegisterPage />);
  });

  it("should render the component", () => {
    expect(RegisterPageComponent).toMatchSnapshot();
  });

  it("should initialize the state correctly", () => {
    const expectedInitialState = {
      name: "",
      username: "",
      email: "",
      password: "",
      organisation:"",
      passwordConfirm: "",
      displayPassword: false,
      hasAgreed: false,
      loading: false,
      registered: false,
      gitLoading: false,
      error: "",
      passwordShown: false,
      passwordChecked: false,
    };

    expect(RegisterPageComponent.state()).toEqual(expectedInitialState);
  });

  it("should toggle the 'hasAgreed' state when called", () => {
    RegisterPageComponent.instance().toggleAgreed();

    expect(RegisterPageComponent.state("hasAgreed")).toBe(true);

    RegisterPageComponent.instance().toggleAgreed();

    expect(RegisterPageComponent.state("hasAgreed")).toBe(false);
  });

  // it("should redirect to the GitHub authentication page when 'toGithubauth' is called", () => {
  //   const mockEvent = {
  //     preventDefault: jest.fn(),
  //   };
  
  //   window.location.href = jest.fn();
  
  //   RegisterPageComponent.instance().toGithubauth();
  
  //   expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
  //   expect(window.location.href).toHaveBeenCalledTimes(1);
  // });
  

  it("should toggle the 'passwordShown' state when 'togglePassword' is called", () => {
    RegisterPageComponent.instance().togglePassword();

    expect(RegisterPageComponent.state("passwordShown")).toBe(true);

    RegisterPageComponent.instance().togglePassword();

    expect(RegisterPageComponent.state("passwordShown")).toBe(false);
  });

  it("should toggle the 'displayPassword' state when 'togglePasswordConfirm' is called", () => {
    RegisterPageComponent.instance().togglePasswordConfirm();

    expect(RegisterPageComponent.state("displayPassword")).toBe(true);

    RegisterPageComponent.instance().togglePasswordConfirm();

    expect(RegisterPageComponent.state("displayPassword")).toBe(false);
  });

  it("should update the state correctly when 'handleOnChange' is called", () => {
    const mockEvent = {
      target: {
        name: "name",
        value: "John Doe",
      },
    };

    RegisterPageComponent.instance().handleOnChange(mockEvent);

    expect(RegisterPageComponent.state("name")).toBe("John Doe");
  });

  it("should validate the email correctly when 'validateEmail' is called", () => {
    const validEmail = "test@example.com";
    const invalidEmail = "invalid-email";

    expect(RegisterPageComponent.instance().validateEmail(validEmail)).toBe(
      true
    );
    expect(RegisterPageComponent.instance().validateEmail(invalidEmail)).toBe(
      false
    );
  });

  it("should update the state correctly and call 'axios.post' when 'handleSubmit' is called with valid data", () => {
    const mockEvent = {
      preventDefault: jest.fn(),
    };

    const axiosPostSpy = jest.spyOn(axios, "post");

    RegisterPageComponent.setState({
      name: "John Doe",
      username: "johndoe",
      email: "test@example.com",
      password: "password",
      passwordConfirm: "password",
      hasAgreed: true,
    });

    RegisterPageComponent.instance().handleSubmit(mockEvent);

    expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    expect(RegisterPageComponent.state("loading")).toBe(true);
    expect(axiosPostSpy).toHaveBeenCalledWith(
      expect.any(String),
      expect.objectContaining({
        name: "John Doe",
        username: "johndoe",
        email: "test@example.com",
        password: "password",
      })
    );
    // You can add more expectations based on the specific behavior you want to test
  });

  it("should update the state correctly when 'handleSubmit' is called with invalid data", () => {
    const mockEvent = {
      preventDefault: jest.fn(),
    };

    RegisterPageComponent.instance().handleSubmit(mockEvent);

    expect(RegisterPageComponent.state("error")).toBe("Please enter all fields");

    // You can add more expectations based on the specific behavior you want to test
  });

  it("should render the registration form when 'registered' state is false", () => {
    RegisterPageComponent.setState({
      registered: false,
    });

    expect(RegisterPageComponent.find(".RegisterContentHeading").exists()).toBe(
      true
    );
    expect(RegisterPageComponent.find("form.LoginContentInputs").exists()).toBe(
      true
    );
    // You can add more expectations based on the specific behavior you want to test
  });

  it("should render the success message when 'registered' state is true", () => {
    RegisterPageComponent.setState({
      registered: true,
      email: "test@example.com",
    });

    expect(RegisterPageComponent.find(".RegisteredMessage").exists()).toBe(true);
    expect(RegisterPageComponent.find(".RegisteredMessage h2").text()).toBe(
      "Thank you for registering with us!"
    );
    expect(RegisterPageComponent.find(".RegisteredMessage p span").text()).toBe(
      "test@example.com"
    );
    // You can add more expectations based on the specific behavior you want to test
  });

  it("should render the GitHub login button", () => {
    RegisterPageComponent.setState({
      gitLoading: false,
    });

    expect(RegisterPageComponent.find(".GithubLoginBtn").exists()).toBe(true);
    expect(RegisterPageComponent.find(".GithubLoginBtn").prop("disabled")).toBe(
      false
    );
    // You can add more expectations based on the specific behavior you want to test
  });
});
