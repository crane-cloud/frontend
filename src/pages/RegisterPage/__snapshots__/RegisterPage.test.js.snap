// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage/> Component should render the component 1`] = `ShallowWrapper {}`;
